backend/stats/views.py

```diff
@@ -7,49 +7,51 @@
 from rest_framework import status
 from django.db.models import Count, Q
 from User.models import User
+from .user_stats_table import UserTable
+from .leader_board_table import LeaderBoardTable
+from .friend_stats_table import FirendsTable
 
+from .mixins import UserTableMixin
+from rest_framework.response import Response
 
-class StatsView(APIView):
+class StatsView(APIView, UserTableMixin):	
 	def get(self, request):
-		x = [1, 2, 3, 4, 5]
-		y = [10, 11, 12, 13, 14]
-		fig = go.Figure(data=[go.Table(header=dict(values=['A Scores', 'B Scores']),
-                 cells=dict(values=[[100, 90, 80, 90], [95, 85, 75, 95]]))
-                     ])
-
-        # Convert plot to HTML
-		plot_html = pio.to_html(fig, full_html=True)
+		user = request.user
+		wins = self.get_wins_count(user)
+		losses = self.get_losses_count(user)
+		total_game_played = self.get_total_games_played_count(user)
+		win_loss_ratio = self.get_win_loss_ratio(user)
+		wining_streak = self.get_win_streak_count(user)
+		position_in_leaderboard = self.get_position_in_leaderboard(user)
+		self.user_table = UserTable(user, wins, losses, total_game_played, win_loss_ratio, wining_streak, position_in_leaderboard)
+        # Create the Plotly figure
+		fig = go.Figure(data=[go.Table(
+			header=dict(values=['Statistic', 'Value']),
+			cells=dict(values=[
+				['Wins', 'Losses', 'Total Games Played', 'Win/Loss Ratio', 'Winning Streak', 'Position in Leaderboard'],
+				[wins, losses, total_game_played, win_loss_ratio, wining_streak, position_in_leaderboard]
+			])
+		)])		
+	
+		# Convert the Plotly figure to an SVG string
+		svg_str = fig.to_image(format="svg").decode("utf-8")
 
+		return HttpResponse(svg_str, content_type='image/svg+xml')
 
-		return HttpResponse(plot_html, content_type='text/html')
-	
-class LeaderBoardView(APIView):
+class LeaderBoardView(APIView, UserTableMixin):
```

Comment:
And be in it's own app

Reply:
I am uncertain about this name change. If the class LeaderBoardView was specifically designed to display a list of leaderboard entries I would have named it so, but it only returns one leaderboard json and not multiple ones. For reference look at class MatchView(APIView) it is named MatchView since it is not listing anything, but we have class FriendsListView(APIView) on the other hand. I changed the name, as requested but still puzzled about what listing means here.